fwoa-tools/src/binaryConverter.ts

```diff
@@ -154,43 +149,51 @@ async function convertBinaryResource(): Promise<void> {
         outputFile.file_names[folderName] = [];
       // eslint-disable-next-line security/detect-object-injection
       outputFile.file_names[folderName].push(newKey);
-      await uploadBinaryResource(newKey, results);
+      await uploadBinaryResource(newKey, results, s3Client);
       logs.write(`${new Date().toISOString()}: Updated Binary .ndjson uploaded to Export Bucket!\n`);
     }
     results = results.trimEnd();
   }
 }
 
-async function checkConfiguration(): Promise<void> {
-  await s3Client.listObjectsV2({ Bucket: exportBucketName! }).promise();
-  console.log('Sucessfully authenticated with S3 Export Bucket...');
-  await s3Client.listObjectsV2({ Bucket: binaryBucketName! }).promise();
-  console.log('Sucessfully authenticated with S3 Binary Bucket...');
-}
-
-async function startBinaryConversion(): Promise<void> {
+async function startBinaryConversion(outputFile: ExportOutput): Promise<void> {
   console.log(`Starting Binary Resource Conversion...`);
   logs.write(`${new Date().toISOString()}: Starting Binary Resource Conversion`);
-  await convertBinaryResource();
+  await convertBinaryResource(outputFile);
   // eslint-disable-next-line security/detect-non-literal-fs-filename
   writeFileSync(`${EXPORT_STATE_FILE_NAME}`, JSON.stringify(outputFile));
   logs.write(`${new Date().toISOString()}: Finished Binary Resource Conversion`);
 }
 
-(async () => {
-  await checkConfiguration();
-  console.log('successfully authenticated to all services');
-  if (!dryRun) {
-    try {
-      await startBinaryConversion();
+async function runScript(): Promise<void> {
+  // eslint-disable-next-line security/detect-non-literal-fs-filename
+  const outputFile = JSON.parse(readFileSync(EXPORT_STATE_FILE_NAME).toString());
+
+  try {
+    await checkConfiguration(logs);
+    console.log('successfully authenticated to all services');
+    if (!dryRun) {
+      await startBinaryConversion(outputFile);
       console.log('successfully converted all binary resources!');
-    } catch (error) {
-      console.log('Failed to process binary resources', error);
-      logs.write(`\n**${new Date().toISOString()}: ERROR!**\n${error}\n`);
     }
+  } catch (error) {
+    console.log('Failed to process binary resources', error);
+    logs.write(`\n**${new Date().toISOString()}: ERROR!**\n${error}\n`);
   }
-  logs.end();
-})().catch((error) => {
-  console.log('some checks failed!', error);
+}
+
+/* istanbul ignore next */
+(async () => {
+  // don't runScript when importing code for unit tests
+  if (!process.argv.includes('test')) {
```

Comment:
We switched to using `if (!process.env.UNIT_TEST) {` since we now pass `UNIT_TEST` as an environment variable when running `rushx test`. Before when we ran `rushx test` we save that it wasn't passing `test` as an argument.